packages/TorneloScoresheet/src/pages/TablePairing/TablePairing.tsx

```diff
@@ -0,0 +1,139 @@
+import React, { useState } from 'react';
+import { useTablePairingState } from '../../context/AppModeStateContext';
+import { ChessGameInfo } from '../../types/ChessGameInfo';
+import { styles } from './style';
+import { Text, TouchableOpacity, View } from 'react-native';
+import Sheet from '../../components/Sheet/Sheet';
+import { chessEngine } from '../../chessEngine/chessEngineInterface';
+import PrimaryButton from '../../components/PrimaryButton/PrimaryButton';
+import { PieceType } from '../../types/ChessMove';
+import { PlayerColour } from '../../types/ChessGameInfo';
+import PieceAsset from '../../components/PieceAsset/PieceAsset';
+import PrimaryText, {
+  FontWeight,
+} from '../../components/PrimaryText/PrimaryText';
+
+const TablePairing: React.FC = () => {
+  const tablePairingState = useTablePairingState();
+  const tablePairingMode = tablePairingState?.[0];
+  const [showSheet, setShowSheet] = useState(false);
+  const [selectedPlayer, setSelectedPlayer] = useState(0);
+  const goToRecording = tablePairingState?.[1].goToRecording;
+
+  //TO DO: LIMIT LENGTH OF INPUT NAME
+  const setPlayer = (playerNumber: number) => {
+    setSelectedPlayer(playerNumber);
+  };
+
+  const handleConfirm = () => {
+    if (!tablePairingMode || !goToRecording) {
+      return;
+    }
+
+    const [board, fen] = chessEngine.startGame();
+    goToRecording(
+      tablePairingMode.pairing,
+      [{ moveNo: 1, whitePly: { startingFen: fen } }],
+      board,
+    );
+  };
+
+  const handleCancel = () => {
+    setShowSheet(false);
+  };
+
+  const displayPlayer = (
```

Comment:
harry has explained to me that all 'html' should be rendered in the final return on a component, so we shouldnt do stuff like this ( a method called displayPlayer that returns 'html'. If you find yourself wanting to call a function in the html, it should probably be its own separate component `PlayerCard`